### DISCUSSION TITLE
A dynamic response at root

### DISCUSSION THREAD

**Discussion by Quafadas** [disneystreaming/smithy4s#542] [Q&A] [answered]
I'm trying to wrap an API, which looks something like this; 

```
GET {{url}}/reverselookupset?filter=BSD90F217,BRTAKH528
``` 
The response, looks something along these lines
```
{
  "BSD90F217": [
    "15faed49-10bf-4325-8914-004424947f5a"
  ],
  "BRTAKH528": [
    "ed885474-f73e-4c1a-9b7c-071c1264779a"
  ]
}
```
I gotta say, that this doesn't look like the way I would choose to model something, but here we are... currently, I think that this is not modelable in smithy - the return type is a 

```
map ReverseLookupMap {
    key: String,
    value: Something
}
list Something {
    member: String    
}
```
To my eye, but the map is at the root of the document. I don't think smithy allows that? As I don't know the field names... I'm stumped... is there a trick to get around this, or is it fairly fundamental?

**Comment by daddykotex** [comment]
Give me a minute to test my hopethises but I think you can!

**Comment by daddykotex** [comment, marked as answer]
You can do something like that:

```kotlin
structure GreetingOutput {
  @required
  @httpPayload
  payload: ReverseLookupMap
}

map ReverseLookupMap {
    key: String,
    value: Something
}
list Something {
    member: String    
}
```

Where `GreetingOutput` is the output of your operation

**Reply by Quafadas** [reply]
FTW! Tested and works great - many thanks... that saved me a lot of pain :-).
